runtime/sema/check_pragma.go

```diff
@@ -0,0 +1,49 @@
+/*
+ * Cadence - The resource-oriented smart contract programming language
+ *
+ * Copyright 2019-2020 Dapper Labs, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package sema
+
+import "github.com/onflow/cadence/runtime/ast"
+
+func (checker *Checker) VisitPragmaDeclaration(p *ast.PragmaDeclaration) ast.Repr {
+
+	invocPragma, isInvocPragma := p.Expression.(*ast.InvocationExpression)
+	_, isIdentPragma := p.Expression.(*ast.IdentifierExpression)
```

Comment:
Slight optimization: Maybe only check if the expression is an identifier expression if it is  not known to be an invocation expression

```suggestion
	var isIdentPragma bool
	if !isInvocPragma {
		_, isIdentPragma = p.Expression.(*ast.IdentifierExpression)
	}
```